tiatoolbox/dataloader/wsireader.py

```diff
@@ -1131,16 +1155,18 @@ def read_bounds(self, bounds, resolution=1.0, units="baseline"):
             resolution=resolution,
             units=units,
         )
-        start_x, start_y, end_x, end_y = bounds
         stride = 2 ** read_level
 
-        im_region = self.img[start_y:end_y:stride, start_x:end_x:stride]
+        im_region = utils.image.safe_padded_read(
+            self.img, bounds, pad_mode="constant", constant_values=255
+        )
+        im_region = utils.transforms.imresize(img=im_region, scale_factor=stride)
```

Comment:
Related to above comment about kwargs

Reply:
Same as above, I will make a new PR for this.